pkg/db/driver_factory.go

```
@@ -5,21 +5,30 @@ import (
 	"fmt"
 
 	"github.com/golang-migrate/migrate/v4/database"
+	"github.com/justtrackio/gosoline/pkg/log"
 )
 
-type DriverFactory interface {
+type DriverFactory func(logger log.Logger) (Driver, error)
+
+type Driver interface {
 	GetDSN(settings Settings) string
 	GetMigrationDriver(db *sql.DB, database string, migrationsTable string) (database.Driver, error)
 }
 
 var connectionFactories = map[string]DriverFactory{}
 
-func GetDriverFactory(driverName string) (DriverFactory, error) {
-	factory, exists := connectionFactories[driverName]
+func GetDriver(logger log.Logger, driverName string) (Driver, error) {
+	var ok bool
+	var err error
+	var factory DriverFactory
+	var driver Driver
 
-	if !exists {
+	if factory, ok = connectionFactories[driverName]; !ok {
 		return nil, fmt.Errorf("no driver factory defined for %s", driverName)
 	}
 
-	return factory, nil
+	if driver, err = factory(logger); err != nil {
+	}
```

do we want to return an error here?

---

:+1: